### DISCUSSION TITLE
Project root folder path variable

### DISCUSSION THREAD

**Discussion by andreg** [zed-industries/zed#12242] [Feature Requests]
Hi,
I was wondering if something along the lines of `{buffer_path}` is available to retrieve the project root folder. Ideally I'd be looking to use it in an external formatter configuration.

**Comment by SomeoneToIgnore** [comment]
If it's the Zed task variables you were referring to, then `ZED_WORKTREE_ROOT` looks like the needed one: https://zed.dev/docs/tasks#variables

**Comment by andreg** [comment]
I'd like the variable to work in a formatter configuration, like so:

<img width="1759" alt="Screenshot 2024-05-24 alle 17 05 59" src="https://github.com/zed-industries/zed/assets/39385/5091c131-b18d-4878-ba44-c52f7b120142">

but apparently, the variable is computed as an empty string.

**Reply by SomeoneToIgnore** [reply]
Yes, the formatter settings are not about tasks (but maybe it should), only tasks get the special variables set, as the docs imply.

**Comment by ahgraber** [comment]
+1 for having the git root / project root available as a variable; would be very nice for indicating/specifying configuration files
